src/app/todo.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { TodoModel, priorityType } from './models/todo-model'
+import { Variable } from '@angular/compiler/src/render3/r3_ast';
+import todoSamples from '../assets/todo-samples.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TodoService {
+
+  todos: TodoModel[] = todoSamples;
+
+  constructor() { 
+
+  }
+
+  addTodo(todo: TodoModel) {
+    this.todos.push(todo);
+  }
+
+  testAddTodo() {
+    let todo = new TodoModel();
+    todo.title = "Get groceries";
+    todo.description = "Go to the store";
+    todo.dateAdded = 319480923.0;
+    todo.deadline = 0;
+    todo.isDone = false;
+    todo.priority = 0;
+    this.todos.push(todo);
+  }
+
+  getAllTodos() {
+    return this.todos;
+  }
+
+  deleteTodo(todo: TodoModel) {
+    this.todos = this.todos.filter(x => x !== todo);
+  }
+
+  toggleCompleted(todo: TodoModel) {
+    this.todos.forEach(x => {
+      if(x === todo)
+      {
+        x.isDone = !x.isDone;
+      }
+    });
+  }
+
+  editTodo(todo: TodoModel, variable: string, edit: any) {
+    this.todos.forEach(x => {
+      if(x === todo)
+      {
+        if(variable === "title")
+        {
+          x.title = edit;
+        }
+        else if(variable === "description")
+        {
+          x.description = edit;
+        }
+        else if(variable === "deadline")
+        {
+          x.deadline = edit;
+        }
+        else if(variable === "priority")
+        {
+          x.priority = edit;
+        }
+      }
+    });
+  }
+
```

Comment:
Sort logic such as sorting should be handled by the respective component and not necessarily the service.